packages/ui/src/hooks/useGlobalShortcut.ts

```diff
@@ -0,0 +1,61 @@
+"use client";
+
+import { useEffect } from "react";
+
+export type Shortcut = {
+  metaKey?: boolean;
+  shiftKey?: boolean;
+  key: string;
+};
+
+function eventMatchesShortcut(event: KeyboardEvent, shortcut: Shortcut) {
+  if (
+    (shortcut.shiftKey && !event.shiftKey) ||
+    (shortcut.metaKey && !event.metaKey) ||
+    event.key.toLowerCase() !== shortcut.key.toLowerCase()
+  ) {
+    return false;
+  }
+  return true;
+}
+
+export function useGlobalShortcut(shortcut: Shortcut, act: () => void) {
```

Comment:
I could remove this, as we have "useGlobalShortcut(s)" now, but I could imagine times when this would be a (bit) better (especially if ``useGlobalShortcut`` always needs memoization) . Not sure though.